camera_calibration/src/camera_calibration/calibrator.py

```diff
@@ -647,19 +661,28 @@ def cal_fromcorners(self, good):
 
         ipts = [ points for (points, _) in good ]
         opts = self.mk_object_points(boards)
-
         # If FIX_ASPECT_RATIO flag set, enforce focal lengths have 1/1 ratio
         intrinsics_in = numpy.eye(3, dtype=numpy.float64)
-        reproj_err, self.intrinsics, dist_coeffs, rvecs, tvecs = cv2.calibrateCamera(
-                   opts, ipts,
-                   self.size,
-                   intrinsics_in,
-                   None,
-                   flags = self.calib_flags)
-        # OpenCV returns more than 8 coefficients (the additional ones all zeros) when CALIB_RATIONAL_MODEL is set.
-        # The extra ones include e.g. thin prism coefficients, which we are not interested in.
-        self.distortion = dist_coeffs.flat[:8].reshape(-1, 1)
 
+        if "pinhole" in self.distortion_model:
+            reproj_err, self.intrinsics, dist_coeffs, rvecs, tvecs = cv2.calibrateCamera(
+                       opts, ipts,
+                       self.size,
+                       intrinsics_in,
+                       None,
+                       flags = self.calib_flags)
+            # OpenCV returns more than 8 coefficients (the additional ones all zeros) when CALIB_RATIONAL_MODEL is set.
+            # The extra ones include e.g. thin prism coefficients, which we are not interested in.
+            self.distortion = dist_coeffs.flat[:8].reshape(-1, 1)
+        elif "fisheye" in self.distortion_model:
+            calibration_flags = cv2.fisheye.CALIB_FIX_SKEW + self.calib_flags # Add user flags
+            reproj_err, self.intrinsics, self.distortion, rvecs, tvecs = cv2.fisheye.calibrate(
+                     opts, ipts,
+                      self.size,
+                      intrinsics_in, None, flags = calibration_flags,
+                      criteria = (cv2.TERM_CRITERIA_EPS+cv2.TERM_CRITERIA_MAX_ITER, 30, 1e-6))
```

Comment:
There's some hard coded numbers, should these be calib flags?

Reply:
There are pinhole and fisheye specific calibration flags. Any suggestion how to deal with this? Ideally we would need to offer the user the ability to choose flags

Reply:
Default criteria flags do provide good convergence, so not including them as seen in last commit.